packages/x-data-grid/src/components/panel/filterPanel/GridFilterInputBoolean.tsx

```diff
@@ -16,6 +16,16 @@ export type GridFilterInputBooleanProps = GridFilterInputValueProps &
     isFilterActive?: boolean;
   };
 
+export const sanitizeFilterItemValue = (value: any): boolean | undefined => {
+  if (String(value).toLowerCase() === 'true') {
+    return true;
+  }
+  if (String(value).toLowerCase() === 'false') {
```

Comment:
do we need this condition? can we simply return false if it is not true or I missed something ?